kinesis/consumer.py

```
@@ -675,6 +686,78 @@ def get_shard_status(self):
+        if ready_fut in pending:
+            ready_fut.cancel()
+
+        if ready_fut in done:
+            return  # Consumer is ready
+
+        if self.fetch_task in done:
+            # Fetch task died or exited before we became ready
+            try:
+                exc = self.fetch_task.exception()
+            except asyncio.CancelledError:
+                raise RuntimeError("Fetch task was cancelled before consumer became ready")
```

Ruff (0.15.5)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)